### ISSUE TITLE
Cannot use `aliases` on `enum` variants

### ISSUE THREAD

**Issue by amountainram** [juhaku/utoipa#1052]
Hi,
I successfully used the `aliases` macro attribute on structs to generate openapi schemas. Then I attempted to do the same with enums with generics like the following:

```rust
#[derive(ToSchema)]
struct MyStruct;

#[derive(Debug, Serialize, Deserialize, ToSchema)]
#[aliases(MyGenericEnumEnum = MyGenericEnum<MyStruct>)]
pub enum MyGenericEnum<T> {
    Variant(T),
}
```

it appears that the resulting json schema does not take care of the T generic (the generic is present in the output json).

This is a snippet to try out the current behavior (which is consistent across v4.2.3 to v5.0.0-beta.0)

```rust
use serde::{Deserialize, Serialize};
use utoipa::{OpenApi, ToSchema};

#[derive(OpenApi)]
#[openapi(components(schemas(MyGenericStruct, MyStruct, MyGenericEnumEnum)))]
pub struct ApiDoc;

#[derive(ToSchema)]
struct MyStruct;

#[derive(Debug, Serialize, Deserialize, ToSchema)]
#[aliases(MyGenericStruct = MyGeneric<MyStruct>)]
pub struct MyGeneric<T> {
    pub generic: Vec<T>,
}

#[derive(Debug, Serialize, Deserialize, ToSchema)]
#[aliases(MyGenericEnumEnum = MyGenericEnum<MyStruct>)]
pub enum MyGenericEnum<T> {
    Variant(T),
}

#[cfg(test)]
mod tests {
    use super::ApiDoc;
    use serde_json::json;
    use utoipa::OpenApi as _;

    #[test]
    fn wrong_json_schema_test() {
        let expected = json!({
          "schemas": {
            "MyGenericEnumEnum": {
              "oneOf": [
                {
                  "type": "object",
                  "required": [
                    "Variant"
                  ],
                  "properties": {
                    "Variant": {
                      "$ref": "#/components/schemas/T"
                    }
                  }
                }
              ]
            },
            "MyGenericStruct": {
              "type": "object",
              "required": [
                "generic"
              ],
              "properties": {
                "generic": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/MyStruct"
                  }
                }
              }
            },
            "MyStruct": {
              "default": null
            }
          }
        });
        let json_schema = serde_json::to_value(
            ApiDoc::openapi()
                .components
                .expect("components to be generated"),
        )
        .expect("cannot generate json schema");
        assert_eq!(json_schema, expected);
    }
}
```
Obviously the test **should not** pass.
any idea why the generic is not erased in the resulting json? is this a bug or I've misused macro attributes?

Thanks for the awesome lib BTW 🙏🏻

**Comment by juhaku** [comment]
This is solved in #1034 which will change the way the generics work in general in `utoipa`. This has not yet been released and is in master at the moment, but will ultimately land in `5.0.0` once it is out.

**Comment by amountainram** [comment]
That's great!
Thanks for the quick support!

Will look forward for the release
